deferred.ts

```diff
@@ -54,21 +54,21 @@ export class DeferredStack<T> {
 
   async pop(): Promise<T> {
     if (this._array.length > 0) {
-      return this._array.pop();
+      return this._array.pop()!;
```

Comment:
Why are `!` needed in methods of `DeferredStack`? All fields are assigned in constructor, it shouldn't need additional assertion.

Reply:
the type of pop in the typescript definition is <T|undefined>, the compiler isn't smart enough to see the guard clause on line 56 and know that it ensures a  <T>, so the choices are that ! or longer shenanigans to make the compiler happy.